Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -442,27 +442,24 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         completion: @escaping (Result<EvaluationResult, Error>) -> Void
     ) throws {
         do {
-            if localFileSystem.isFile(manifestPath) {
+            try withTemporaryFile(suffix: ".swift") { tempFile, cleanupTempFile in
+                try localFileSystem.writeFileContents(tempFile.path, bytes: ByteString(manifestContents))
                 try self.evaluateManifest(
-                    at: manifestPath,
+                    at: tempFile.path,
                     packageIdentity: packageIdentity,
                     toolsVersion: toolsVersion,
-                    delegateQueue:  delegateQueue,
-                    callbackQueue: callbackQueue,
-                    completion: completion
-                )
-            } else {
-                try withTemporaryFile(suffix: ".swift") { tempFile, cleanupTempFile in
-                    try localFileSystem.writeFileContents(tempFile.path, bytes: ByteString(manifestContents))
-                    try self.evaluateManifest(
-                        at: tempFile.path,
-                        packageIdentity: packageIdentity,
-                        toolsVersion: toolsVersion,
-                        delegateQueue: delegateQueue,
-                        callbackQueue: callbackQueue
-                    ) { result in
-                        dispatchPrecondition(condition: .onQueue(callbackQueue))
-                        cleanupTempFile(tempFile)
+                    delegateQueue: delegateQueue,
+                    callbackQueue: callbackQueue
+                ) { result in
+                    dispatchPrecondition(condition: .onQueue(callbackQueue))
+                    cleanupTempFile(tempFile)
+
+                    switch result {
+                    case .success(var evaluationResult):
+                        // Replace the temporary path with the orgiginal one in compiler output.
+                        evaluationResult.compilerOutput = evaluationResult.compilerOutput?.replacingOccurrences(of: tempFile.path.pathString, with: manifestPath.pathString)
```

Comment:
This seems fairly hacky TBH.  It would be less hacky (though a lot more complicated) to use a compiler VFS (assuming Swift supports that, as Clang does).  In particular because this doesn't take into account any paths in the structured diagnostics file at all, right?

Reply:
Looks like it does support `-vfsoverlay`, agreed that it seems like a better solution.